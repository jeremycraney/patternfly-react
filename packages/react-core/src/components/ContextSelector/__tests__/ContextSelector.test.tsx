import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { shallow } from 'enzyme';
import { ContextSelector } from '../ContextSelector';
import { ContextSelectorItem } from '../ContextSelectorItem';

const items = [
  <ContextSelectorItem key="0">My Project</ContextSelectorItem>,
  <ContextSelectorItem key="1">OpenShift Cluster</ContextSelectorItem>,
  <ContextSelectorItem key="2">Production Ansible</ContextSelectorItem>,
  <ContextSelectorItem key="3">AWS</ContextSelectorItem>,
  <ContextSelectorItem key="4">Azure</ContextSelectorItem>
];

test('Renders ContextSelector', () => {
  const view = render(<ContextSelector> {items} </ContextSelector>);
  expect(view.container).toMatchSnapshot();
});

test('Renders ContextSelector open', () => {
  const view = shallow(<ContextSelector isOpen> {items} </ContextSelector>);
  expect(view).toMatchSnapshot();
});

test('Verify onToggle is called ', () => {
  const mockfn = jest.fn();
  render(<ContextSelector onToggle={mockfn}> {items} </ContextSelector>);

  userEvent.click(screen.getByRole('button'));
  expect(mockfn.mock.calls).toHaveLength(1);
});
