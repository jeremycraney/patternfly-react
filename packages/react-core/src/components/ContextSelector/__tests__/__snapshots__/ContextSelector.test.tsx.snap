// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ContextSelector 1`] = `
<div>
  <div
    class="pf-c-context-selector"
    data-ouia-component-id="OUIA-Generated-ContextSelector-1"
    data-ouia-component-type="PF4/ContextSelector"
    data-ouia-safe="true"
  >
    
    <button
      aria-expanded="false"
      aria-labelledby="pf-context-selector-label-id-0 pf-context-selector-toggle-id-0"
      class="pf-c-context-selector__toggle"
      id="pf-context-selector-toggle-id-0"
      type="button"
    >
      <span
        class="pf-c-context-selector__toggle-text"
      />
      <span
        class="pf-c-context-selector__toggle-icon"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          height="1em"
          role="img"
          style="vertical-align: -0.125em;"
          viewBox="0 0 320 512"
          width="1em"
        >
          <path
            d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;

exports[`Renders ContextSelector open 1`] = `
<div
  className="pf-c-context-selector pf-m-expanded"
  data-ouia-component-id="OUIA-Generated-ContextSelector-2"
  data-ouia-component-type="PF4/ContextSelector"
  data-ouia-safe={true}
>
  <ContextSelectorToggle
    aria-labelledby="pf-context-selector-label-id-0 pf-context-selector-toggle-id-0"
    className=""
    id="pf-context-selector-toggle-id-0"
    isActive={false}
    isOpen={true}
    isPlain={false}
    isText={false}
    onEnter={[Function]}
    onToggle={[Function]}
    parentRef={null}
    toggleText=""
  />
  <div
    className="pf-c-context-selector__menu"
  >
    <FocusTrap
      active={true}
      focusTrapOptions={
        Object {
          "clickOutsideDeactivates": true,
        }
      }
      paused={false}
      preventScrollOnDeactivate={false}
    >
      <div
        className="pf-c-context-selector__menu-search"
      >
        <InputGroup>
          <TextInput
            aria-labelledby="pf-context-selector-search-button-id-0"
            onChange={[Function]}
            onKeyPress={[Function]}
            placeholder="Search"
            type="search"
            value=""
          />
          <Button
            aria-label="Search menu items"
            id="pf-context-selector-search-button-id-0"
            onClick={[Function]}
            variant="control"
          >
            <SearchIcon
              aria-hidden="true"
              color="currentColor"
              noVerticalAlign={false}
              size="sm"
            />
          </Button>
        </InputGroup>
      </div>
      <ContextProvider
        value={
          Object {
            "onSelect": [Function],
          }
        }
      >
        <ContextSelectorMenuList
          className=""
          isOpen={true}
        >
           
          <ContextSelectorItem
            className=""
            href={null}
            isDisabled={false}
            key="0"
            onClick={[Function]}
            sendRef={[Function]}
          >
            My Project
          </ContextSelectorItem>
          <ContextSelectorItem
            className=""
            href={null}
            isDisabled={false}
            key="1"
            onClick={[Function]}
            sendRef={[Function]}
          >
            OpenShift Cluster
          </ContextSelectorItem>
          <ContextSelectorItem
            className=""
            href={null}
            isDisabled={false}
            key="2"
            onClick={[Function]}
            sendRef={[Function]}
          >
            Production Ansible
          </ContextSelectorItem>
          <ContextSelectorItem
            className=""
            href={null}
            isDisabled={false}
            key="3"
            onClick={[Function]}
            sendRef={[Function]}
          >
            AWS
          </ContextSelectorItem>
          <ContextSelectorItem
            className=""
            href={null}
            isDisabled={false}
            key="4"
            onClick={[Function]}
            sendRef={[Function]}
          >
            Azure
          </ContextSelectorItem>
           
        </ContextSelectorMenuList>
      </ContextProvider>
    </FocusTrap>
  </div>
</div>
`;
